src/system/AdminModule/Controller/AdminInterfaceController.php

```diff
@@ -0,0 +1,773 @@
+    			$html .= ' class="'.$active.$dropdown;
+    			$html .= !empty($class) ? $class : '';
+    			$html .= '">';
+    			$attr  = !empty($menuitem['title']) ? ' title="'.$menuitem['title'].'"' : '';
+    			$attr .= !empty($menuitem['class']) ? ' class="'.$menuitem['class'].'"' : '';
+    	
+    			if (isset($menuitem['disabled']) && $menuitem['disabled'] == true) {
+    				$html .= '<a '.$attr.'>'.$menuitem['text'].'</a>';
+    			} elseif (!empty($menuitem['url'])) {
+    				$icon = '';
+    				if (!empty($menuitem['icon'])) {
+    					$icon = '<span class="fa fa-'.$menuitem['icon'].'"></span> ';
+    				}
+    				$html .= '<a href="'.DataUtil::formatForDisplay($menuitem['url']).'"'.$attr.' style="display: inline-block;">'.$icon.$menuitem['text'].'</a>';
+    				if (isset($menuitem['links'])) {
+    					$html .= '<a href="#" class="dropdown-toggle" data-toggle="dropdown" style="text-decoration: none;">&nbsp;<b class="caret"></b></a>';
+    				}
+    			} else {
+    				$html .= '<span'.$attr.'>'.$menuitem['text'].'</span>';
+    			}
+    			if (isset($menuitem['links'])) {
+    				$html .= '<ul class="dropdown-menu">';
+    				foreach($menuitem['links'] as $submenuitem) {
+    					$html .= '<li>';
+    					if (isset($submenuitem['url'])) {
+    						$html .= '<a href="'.DataUtil::formatForDisplay($submenuitem['url']).'">'.$submenuitem['text'].'</a>';
+    					} else {
+    						$html .= $submenuitem['text'];
+    					}
+    					$html .= '</li>';
+    				}
+    				$html .= '</ul>';
+    			}
+    			$html .= '</li>';
+    		}
+    	
+    		$html .= '</ul>';
+    	}    	
+    	
+    	return ['menu' => $html];
+    }    
+    
+    
+    /**
+     * Get security analyzer data.
+     *
+     * @return array data
+     */
+    private function _securityanalyzer()
+    {
+        $data = array();
+
+        // check for magic_quotes
+        $data['magic_quotes_gpc'] = DataUtil::getBooleanIniValue('magic_quotes_gpc');
+
+        // check for register_globals
+        $data['register_globals'] = DataUtil::getBooleanIniValue('register_globals');
+
+        // check for config.php being writable
+        $data['config_php'] = (bool)is_writable('config/config.php');
+
+        // check for .htaccess in temp directory
+        $app_htaccess = false;
+        $appDir = $this->get('kernel')->getRootDir();
+        if ($appDir) {
+            // check if we have an absolute path which is possibly not within the document root
+            $docRoot = System::serverGetVar('DOCUMENT_ROOT');
+            if (StringUtil::left($appDir, 1) == '/' && (strpos($appDir, $docRoot) === false)) {
+                // temp dir is outside the webroot, no .htaccess file needed
+                $app_htaccess = true;
+            } else {
+                if (strpos($appDir, $docRoot) === false) {
+                    $ldir = dirname(__FILE__);
+                    $p = strpos($ldir, DIRECTORY_SEPARATOR.'system'); // we are in system/Admin
```

Comment:
refactor variable names to have more meaning - e.g. `$p` has no meaning. This makes the code difficult to read for later developers.